independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/AnnotationsTransformation.java

```diff
@@ -46,6 +46,7 @@ public interface AnnotationsTransformation<T extends AnnotationsTransformation<T
      */
     T add(Class<? extends Annotation> annotationType, AnnotationValue... values);
 
+    // TODO this isn't really right for IP transformations anymore?
```

Comment:
If so what should we do about it?

Reply:
Ah, I completely forgot about this, good point! Let me refresh my memory on what was the issue here...

Reply:
You can still use what the javadoc says. In fact you'll have to if you don't migrate to new APIs and if you do this for observer methods.
However, we can change the javadoc to be more precise:

     * NOTE: The annotation will belong to an annotation target which is derived from the transformation context.
     * Each transformation context has means to obtain its annotation target such as
     * {@link AnnotationsTransformer.TransformationContext#getTarget()} or
     * {@link InjectionPointsTransformer.TransformationContext#getAnnotationTarget()}.
     * <p>
     * If you need to add an annotation instance to a method parameter and the annotation target is the whole method, use
     * methods consuming {@link AnnotationInstance} directly and supply the correct {@link MethodParameterInfo}.

Reply:
Applied the above wording and removed the `TODO`.